controller/controller.go

```diff
@@ -260,6 +300,68 @@ func (c controller) GetStatus(req Request) ([]fleet.UnitStatus, error) {
 	return status, maskAny(err)
 }
 
+func (c controller) WaitForStatus(req Request, desired Status, closer <-chan struct{}) error {
+	fail := make(chan error)
+	done := make(chan struct{})
+
+	go func() {
+		// c describes the count of how often the desired aggregated status was
+		// seen.
+		count := 0
+
+	L1:
+		for {
+			unitStatusList, err := c.groupStatus(req)
+			if IsUnitNotFound(err) && desired == StatusNotFound {
```

Comment:
Note that `groupStatus` returns this error as soon as one slice is not found, which might not be the behavior we want here.